### PR TITLE
make bug_report_model minimal

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

An extremely minimal Lightning script. No license, docstrings, doctests, ads etc.
This is meant for
- quickly sharing reproducible code and Trainer settings
- quickly test and develop an idea
- easy copy-paste to and from GitHub/Slack

**Why not use pl_examples/bug_report_model.py?**
Because bug_report_model.py is meant for users. It has plenty of instructions that are not desired for local development and sharing. It unfortunately also contains some method/class names that make PyCharm believe it is a unittest file.
We need an uncomplicated, easy to use script for internal use.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs) SLACK INTERNAL
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pl_examples/boring_script.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    def test_step(self, batch, batch_idx):
+        loss = self(batch).sum()
+        self.log("test_loss", loss)
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.layer.parameters(), lr=0.1)
+
+
+def run():
+    train_data = torch.utils.data.DataLoader(RandomDataset(32, 64), batch_size=2, num_workers=0)
```
Comment (kaushikb11) [inline pl_examples/boring_script.py:47]: Don't think it is necessary to add `num_workers` argument. The default is 0.

**Review by kaushikb11 — APPROVED** [review]
LGTM! It's my new best friend lol

File: pl_examples/boring_script.py (hunk 1 above)
Comment (awaelchli) [reply pl_examples/boring_script.py]: True :)

**Comment by carmocca** [comment]
Why not delete pl_examples/bug_report_model.py?

The boring model notebook is meant for users.

IMO no point keeping this and that one

File: pl_examples/boring_script.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        self.log("test_loss", loss)
+
+    def configure_optimizers(self):
+        return torch.optim.SGD(self.layer.parameters(), lr=0.1)
+
+
+def run():
+    train_data = torch.utils.data.DataLoader(RandomDataset(32, 64), batch_size=2)
+    val_data = torch.utils.data.DataLoader(RandomDataset(32, 64), batch_size=2)
+    test_data = torch.utils.data.DataLoader(RandomDataset(32, 64), batch_size=2)
```
Comment (carmocca) [inline pl_examples/boring_script.py:49]:
```suggestion
    train_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    val_data = DataLoader(RandomDataset(32, 64), batch_size=2)
    test_data = DataLoader(RandomDataset(32, 64), batch_size=2)
```

File: pl_examples/boring_script.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
+import os
+import torch
+from torch.utils.data import Dataset
```
Comment (carmocca) [inline pl_examples/boring_script.py:3]:
```suggestion
from torch.utils.data import DataLoader, Dataset
```

**Review by carmocca — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

**Review by Borda — APPROVED** [review]
